### ISSUE TITLE
Fix casting for C++ compiling

### ISSUE THREAD

**Issue by roflson** [intel/intel-cmt-cat#66]
Hi,

Please could we add casting in **pqos.h** for the casts for these on line 1187:

```
        uint64_t * const p_64 = value;
        double * const p_dbl = value;
```
to
```
        uint64_t * const p_64 = (uint64_t *)value;
        double * const p_dbl = (double* )value;
```

This fails to compile with a C++ compiler without -fpermissive

Sorry this isn't a pull request, i'm behind a convoluted firewall which makes it difficult.

Thanks,
Paul

**Comment by mstarzyx** [comment]
Fix is pushed and will be included in 2.0 release.

**Comment by roflson** [comment]
Great, thanks!
